packages/app/background-jobs-common/src/background-job-processor/utils.ts

```diff
@@ -16,3 +20,27 @@ export const isStalledJobError = (error: Error): boolean =>
 export const sanitizeRedisConfig = (config: RedisConfig): RedisConfig => {
   return { ...config, keyPrefix: undefined }
 }
+
+export const isRedisClient = (redis: RedisConfig | Redis): redis is Redis => 'options' in redis
+
+export const prepareJobOptions = <JobOptionsType extends JobsOptions>(
+  isTest: boolean,
+  options?: JobOptionsType,
+): JobOptionsType => {
+  const preparedOptions: JobOptionsType = {
+    jobId: generateMonotonicUuid(),
+    ...DEFAULT_JOB_CONFIG,
+    ...(options ?? ({} as JobOptionsType)),
+  }
+
+  if (isTest && typeof preparedOptions.backoff === 'object') {
+    preparedOptions.backoff.delay = 1
```

Comment:
might be helpful to add comment, explaining that this is done because zero delay is handled weirly in BullMQ for concurrent job.

Reply:
You are absolutely right! will add it :D